httptools/parser/parser.pyx

```diff
@@ -95,19 +95,24 @@ cdef class HttpParser:
         self._last_error = None
 
     cdef _maybe_call_on_header(self):
-        if self._current_header_name is not None:
-            current_header_name = self._current_header_name
-            current_header_value = self._current_header_value
+        current_header_name = self._current_header_name
+        current_header_value = self._current_header_value
 
-            self._current_header_name = self._current_header_value = None
+        self._current_header_name = self._current_header_value = None
 
-            if self._proto_on_header is not None:
-                self._proto_on_header(current_header_name,
-                                      current_header_value)
+        if self._proto_on_header is not None:
+            self._proto_on_header(current_header_name,
+                                  current_header_value)
 
     cdef _on_header_field(self, bytes field):
-        self._maybe_call_on_header()
-        self._current_header_name = field
+        if self._current_header_name is None:
```

Comment:
Not sure about the `if/elif/else` here.
Why not just calling `_maybe_call_on_header` if `_current_header_name` is `None` (which means we are starting a new header) ?

Reply:
updated